dissect/target/plugins/general/osinfo.py

```diff
@@ -0,0 +1,39 @@
+from typing import Callable, Generator
+
+from flow.record import GroupedRecord, Record
+
+from dissect.target import plugin
+from dissect.target.exceptions import PluginError
+from dissect.target.helpers.record import TargetRecordDescriptor
+
+OSInfoRecord = TargetRecordDescriptor(
+    "generic/osinfo",
+    [
+        ("string", "name"),
+        ("string", "value"),
+    ],
+)
+
+
+class OSInfoPlugin(plugin.Plugin):
+    """Convenience plugin that wraps _os.* functions in records."""
+
+    def check_compatible(self) -> bool:
+        return True
+
+    @plugin.export(record=OSInfoRecord)
+    def osinfo(self) -> Generator[Record, None, None]:
+        for os_func in self.target._os.__functions__:
+            if os_func in ["is_compatible", "get_all_records"]:
+                continue
+            value = getattr(self.target._os, os_func)
+            record = OSInfoRecord(name=os_func, value=repr(value), _target=self.target)
```

Comment:
I don't like the `repr(value)`  here as it adds extra quotes to values that don't need them. Looking at the output I noticed a few things that I think we should address:

```
$ target-query -q -f osinfo ./WinDev2007Eval-disk1.vmdk 
<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='architecture' value="'amd64-win64'">
<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='hostname' value="'WINDEV2007EVAL'">
<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='ips' value="['172.17.19.136', '192.168.73.134', '10.0.2.15']">
<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='os' value="'windows'">
<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='users' value='<bound method WindowsPlugin.users of <dissect.target.plugins.os.windows._os.WindowsPlugin object at 0x7d3d6467f0a0>>'>
<generic/osinfo/grouped [<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='users' value='<bound method WindowsPlugin.users of <dissect.target.plugins.os.windows._os.WindowsPlugin object at 0x7d3d6467f0a0>>'>, <windows/user hostname='WINDEV2007EVAL' domain=None sid='S-1-5-18' name='systemprofile' home='%systemroot%\\system32\\config\\systemprofile'>]>
<generic/osinfo/grouped [<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='users' value='<bound method WindowsPlugin.users of <dissect.target.plugins.os.windows._os.WindowsPlugin object at 0x7d3d6467f0a0>>'>, <windows/user hostname='WINDEV2007EVAL' domain=None sid='S-1-5-19' name='LocalService' home='%systemroot%\\ServiceProfiles\\LocalService'>]>
<generic/osinfo/grouped [<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='users' value='<bound method WindowsPlugin.users of <dissect.target.plugins.os.windows._os.WindowsPlugin object at 0x7d3d6467f0a0>>'>, <windows/user hostname='WINDEV2007EVAL' domain=None sid='S-1-5-20' name='NetworkService' home='%systemroot%\\ServiceProfiles\\NetworkService'>]>
<generic/osinfo/grouped [<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='users' value='<bound method WindowsPlugin.users of <dissect.target.plugins.os.windows._os.WindowsPlugin object at 0x7d3d6467f0a0>>'>, <windows/user hostname='WINDEV2007EVAL' domain=None sid='S-1-5-21-2409144726-2152242204-271938582-1001' name='User' home='C:\\Users\\User'>]>
<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='version' value="'Windows 10 Enterprise Evaluation (NT 10.0) 19041.329'">
```

- The separate `<generic/osinfo hostname='WINDEV2007EVAL' domain=None name='users' value='<bound method WindowsPlugin.users of <dissect.target.plugins.os.windows._os.WindowsPlugin object at 0x7d3d6467f0a0>>'>` entry has little value I think.
- The way grouping turns out means there is a lot of duplication.

We can go at it in different ways. One way is to skip the grouping at all and output sub-records verbatim. That does mean the relation to osinfo is lost, but that may be ok.

Another way is to group all sub-records into a single group. This has as a downside that if the amount of subrecords gets very large, memory consumption will also increase. As `users` is currently the only plugin in `_os.py` that outputs records, this may not be a problem.

I think the best `value` for the `generic/osinfo` record in case it is a `Callable` should be empty as the `repr` of some function is not very useful. Also `Callable` can be a bit more refined to `Generator`, `isinstance` will return True for functions that yield values (as is the case with `users()`), but not for lists etc. (as is the case for `ips`).